Meanfield model

garandel opened the pull request:
I would like put this on the meanfield_model branch.
That's working in the first order except that sigma_V is not yet to the sqrtk().
I couldn't add sqrtk() because otherwise I overflowed ITCM.

andrewgait reviewed (changes requested):
This looks in general fine to me; there are a few considerations you can make (see comments) as to where you can avoid some divides.  I don't think these will help hugely with making enough space to use sqrtk() - we may have to think of another way around that.  But I would be happy to merge this once you've addressed the comments and we can discuss what might be possible for sqrtk() after that.

garandel commented:
Hi Andrew,
yes I noticed before sending to the remote that I could do these
modifications, that I did on another branch on my local repo.
The problem is that it overflowed more with multiplication than with
division.
I think that the main reason is because the multiplication selected for
that is a too big memory needed.
So my question is, could I select one particular multiplication without
writing the name of the function and passing to " / "
 symbol?

On the division side I wonder also if I could use divergente series in
order to approximate the result efficiently?
Same idea for the square root but I didn't know any divergente series for
the square root, yet!! ^^

I was fight with this issue so sending this intermediate code was a step in
order to know the procedure.
But now that I know it, I will send it to you when the job is done.

Sincerely

Thomas


Le lun. 29 nov. 2021 à 10:04, Andrew Gait ***@***.***> a
écrit :

> ***@***.**** requested changes on this pull request.
>
> This looks in general fine to me; there are a few considerations you can
> make (see comments) as to where you can avoid some divides. I don't think
> these will help hugely with making enough space to use sqrtk() - we may
> have to think of another way around that. But I would be happy to merge
> this once you've addressed the comments and we can discuss what might be
> possible for sqrtk() after that.
> ------------------------------
>
> In neural_modelling/src/meanfield/models/meanfield_model_impl.c
> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#discussion_r758145102>
> :
>
> > -        + config->P2*(config->sV-config->sV0)/config->DsV0\
> -        + config->P3*(config->TvN-config->TvN0)/config->DTvN0\
> -        + config->P5*((config->muV-config->muV0)/config->DmuV0)*((config->muV-config->muV0)/config->DmuV0)\
> -        + config->P6*((config->sV-config->sV0)/config->DsV0)*((config->sV-config->sV0)/config->DsV0)\
> -        + config->P7*((config->TvN-config->TvN0)/config->DTvN0)*((config->TvN-config->TvN0)/config->DTvN0)\
> -        + config->P8*(config->muV-config->muV0)/config->DmuV0*(config->sV-config->sV0)/config->DsV0\
> -        + config->P9*(config->muV-config->muV0)/config->DmuV0*(config->TvN-config->TvN0)/config->DTvN0\
> -        + config->P10*(config->sV-config->sV0)/config->DsV0*(config->TvN-config->TvN0)/config->DTvN0;
> -        */
> -
> -    config->Vthre = config->P0;
> -
> -    //return config->Vthre;
> +
> +    Vthre = P0\
> +        + P1*(muV-muV0)/DmuV0\
>
> DmuV0, DsV0 and DTvN0 are all parameters that come directly from constant
> values in python - you can avoid the divides here in this function by
> instead sending in 1/DmuV0 etc. and multiplying.
> ------------------------------
>
> In neural_modelling/src/meanfield/models/meanfield_model_impl.c
> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#discussion_r758149127>
> :
>
> > @@ -251,121 +236,153 @@ void get_fluct_regime_varsup(REAL Ve, REAL Vi, config_t *restrict params)
>      {
>          Fi += ACS_DBL_TINY;
>      }
> +
> +    REAL Tv = ( Fe*(Ue*Te)*(Ue*Te) + Fi*(Ti*Ui)*(Ti*Ui))\
> +        /(Fe*(Ue*Te)*(Ue*Te)/(Te+Tm) + Fi*(Ti*Ui)*(Ti*Ui)/(Ti+Tm));
> +
> +    if (Tv < ACS_DBL_TINY){
> +        Tv += ACS_DBL_TINY;
> +    }
> +
> +    pNetwork->TvN = Tv*Gl/Cm; // Thomas : Heu, useless no?? |resp-> TvN is a dimensional so usefull var
>
> Worth considering 1/Cm here too.
> ------------------------------
>
> In neural_modelling/src/meanfield/models/meanfield_model_impl.c
> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#discussion_r758149203>
> :
>
> >
> -    params->muV = (muGe*params->Ee+muGi*params->Ei+params->Gl*params->El); //fake one
> +    pNetwork->muGn = muG/Gl;
>
> Here Gl is a constant, so again sending in 1/Gl and multiplying may help.
> (In this instance I think you probably need to send both Gl and 1/Gl).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#pullrequestreview-817500766>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARUEWUMFBZKS6ARK77DXHMDUOM63BANCNFSM5I2ZLOTA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

garandel commented:
rectification it's " * " symbol and not " / " one sorry.
Also the problem with Cm and Gl is that there exist also formulas with
multiplication, so the two possibilities are keep like this and do the
division or have two more constante with Gl and Cm so 1/Gl and 1/Cm.

Have a nice day

Thomas

Le lun. 29 nov. 2021 à 13:44, thomas garandel ***@***.***> a
écrit :

> Hi Andrew,
> yes I noticed before sending to the remote that I could do these
> modifications, that I did on another branch on my local repo.
> The problem is that it overflowed more with multiplication than with
> division.
> I think that the main reason is because the multiplication selected for
> that is a too big memory needed.
> So my question is, could I select one particular multiplication without
> writing the name of the function and passing to " / "
>  symbol?
>
> On the division side I wonder also if I could use divergente series in
> order to approximate the result efficiently?
> Same idea for the square root but I didn't know any divergente series for
> the square root, yet!! ^^
>
> I was fight with this issue so sending this intermediate code was a step
> in order to know the procedure.
> But now that I know it, I will send it to you when the job is done.
>
> Sincerely
>
> Thomas
>
>
> Le lun. 29 nov. 2021 à 10:04, Andrew Gait ***@***.***> a
> écrit :
>
>> ***@***.**** requested changes on this pull request.
>>
>> This looks in general fine to me; there are a few considerations you can
>> make (see comments) as to where you can avoid some divides. I don't think
>> these will help hugely with making enough space to use sqrtk() - we may
>> have to think of another way around that. But I would be happy to merge
>> this once you've addressed the comments and we can discuss what might be
>> possible for sqrtk() after that.
>> ------------------------------
>>
>> In neural_modelling/src/meanfield/models/meanfield_model_impl.c
>> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#discussion_r758145102>
>> :
>>
>> > -        + config->P2*(config->sV-config->sV0)/config->DsV0\
>> -        + config->P3*(config->TvN-config->TvN0)/config->DTvN0\
>> -        + config->P5*((config->muV-config->muV0)/config->DmuV0)*((config->muV-config->muV0)/config->DmuV0)\
>> -        + config->P6*((config->sV-config->sV0)/config->DsV0)*((config->sV-config->sV0)/config->DsV0)\
>> -        + config->P7*((config->TvN-config->TvN0)/config->DTvN0)*((config->TvN-config->TvN0)/config->DTvN0)\
>> -        + config->P8*(config->muV-config->muV0)/config->DmuV0*(config->sV-config->sV0)/config->DsV0\
>> -        + config->P9*(config->muV-config->muV0)/config->DmuV0*(config->TvN-config->TvN0)/config->DTvN0\
>> -        + config->P10*(config->sV-config->sV0)/config->DsV0*(config->TvN-config->TvN0)/config->DTvN0;
>> -        */
>> -
>> -    config->Vthre = config->P0;
>> -
>> -    //return config->Vthre;
>> +
>> +    Vthre = P0\
>> +        + P1*(muV-muV0)/DmuV0\
>>
>> DmuV0, DsV0 and DTvN0 are all parameters that come directly from constant
>> values in python - you can avoid the divides here in this function by
>> instead sending in 1/DmuV0 etc. and multiplying.
>> ------------------------------
>>
>> In neural_modelling/src/meanfield/models/meanfield_model_impl.c
>> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#discussion_r758149127>
>> :
>>
>> > @@ -251,121 +236,153 @@ void get_fluct_regime_varsup(REAL Ve, REAL Vi, config_t *restrict params)
>>      {
>>          Fi += ACS_DBL_TINY;
>>      }
>> +
>> +    REAL Tv = ( Fe*(Ue*Te)*(Ue*Te) + Fi*(Ti*Ui)*(Ti*Ui))\
>> +        /(Fe*(Ue*Te)*(Ue*Te)/(Te+Tm) + Fi*(Ti*Ui)*(Ti*Ui)/(Ti+Tm));
>> +
>> +    if (Tv < ACS_DBL_TINY){
>> +        Tv += ACS_DBL_TINY;
>> +    }
>> +
>> +    pNetwork->TvN = Tv*Gl/Cm; // Thomas : Heu, useless no?? |resp-> TvN is a dimensional so usefull var
>>
>> Worth considering 1/Cm here too.
>> ------------------------------
>>
>> In neural_modelling/src/meanfield/models/meanfield_model_impl.c
>> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#discussion_r758149203>
>> :
>>
>> >
>> -    params->muV = (muGe*params->Ee+muGi*params->Ei+params->Gl*params->El); //fake one
>> +    pNetwork->muGn = muG/Gl;
>>
>> Here Gl is a constant, so again sending in 1/Gl and multiplying may help.
>> (In this instance I think you probably need to send both Gl and 1/Gl).
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/SpiNNakerManchester/sPyNNaker/pull/1136#pullrequestreview-817500766>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ARUEWUMFBZKS6ARK77DXHMDUOM63BANCNFSM5I2ZLOTA>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>>
>
